dicom_utils/types.py

```diff
@@ -12,31 +13,86 @@
 Dicom = FileDataset
 
 
-def process_image_type(dcm: Dicom) -> Dict[str, Any]:
-    r"""Extracts and processes fields that identify the DICOM image type.
-    The returned dectionary will be populated with the following items (as available):
+class SimpleImageType(Enum):
+    UNKNOWN = 0
+    NORMAL = 1
+    SVIEW = 2
+    TOMO = 3
+
+    @staticmethod
+    def from_dicom(dcm: Dicom) -> "SimpleImageType":
+        img_type = ImageType.from_dicom(dcm)
+        return img_type.to_simple_image_type()
+
+    def __str__(self) -> str:
+        return repr(self)
+
+    def __repr__(self) -> str:
+        if self is self.UNKNOWN:
+            return "unknown"
+        elif self is self.NORMAL:
+            return "2d"
+        elif self is self.SVIEW:
+            return "s-view"
+        elif self is self.TOMO:
+            return "tomo"
+        else:
+            raise RuntimeError("unknown ImageType value")
+
+
+@dataclass(frozen=True)
+class ImageType:
+    """Container for DICOM metadata fields related to the image type.
+
+    Contains the following attributes:
+
         * ``"pixels"`` - First element of the IMAGE_TYPE field.
         * ``"exam"`` - Second element of the IMAGE_TYPE field.
         * ``"flavor"`` - Third element of the IMAGE_TYPE field.
         * ``"extras"`` - Additional IMAGE_TYPE fields if available
         * ``"NumberOfFrames"`` - Frame count (for TOMO images only)
         * ``"model"`` - Manufacturer's model name
     """
-    # http://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.8.24.3.2.html
 
-    result: Dict[str, Any] = {}
-    result["NumberOfFrames"] = dcm.get("NumberOfFrames", 1)
-    if (model := dcm.get(0x00081090, None)) is not None:
-        result["model"] = model.value
-    else:
-        result["model"] = ""
+    pixels: str
+    exam: str
+    flavor: Optional[str] = None
+    extras: Optional[List[str]] = None
+    NumberOfFrames: Optional[int] = None
+    model: Optional[str] = None
+
+    def __bool__(self) -> bool:
+        return bool(self.pixels) and bool(self.exam)
+
+    def __post_init__(self):
+        if (nf := self.NumberOfFrames) is not None and not isinstance(nf, int):
+            object.__setattr__(self, "NumberOfFrames", int(nf))
+        assert self.NumberOfFrames is None or self.NumberOfFrames > 0
+
+    def simple_repr(self) -> str:
+        s = "|".join(x for x in (self.pixels, self.exam))
+        if self.flavor is not None:
+            s += f"|{self.flavor}" if self.flavor else "|''"
+        for elem in self.extras or []:
+            if not elem or elem.isdigit():
+                continue
+            s += f"|{elem}"
+        return s
+
+    @classmethod
+    def from_dicom(cls, dcm: Dicom) -> "ImageType":
+        result: Dict[str, Any] = {}
+        result["NumberOfFrames"] = dcm.get("NumberOfFrames", 1)
```

Comment:
From the above discussion, this should be:
`result["NumberOfFrames"] = dcm.get("NumberOfFrames", None)`